pkg/verify/configmap.go

```diff
@@ -0,0 +1,87 @@
+package verify
+
+import (
+	"bytes"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/openpgp"
+	"k8s.io/klog"
+)
+
+// ReleaseAnnotationConfigMapVerifier is an annotation set on a config map in the
+// release payload to indicate that this config map controls signing for the payload.
+// Only the first config map within the payload should be used, regardless of whether
+// it has data. See NewFromConfigMap for more.
+const ReleaseAnnotationConfigMapVerifier = "release.openshift.io/verification-config-map"
+
+// NewFromConfigMap expects to receive the data field of the first config map in the release
+// image payload with the annotation "release.openshift.io/verification-config-map". Only the
+// first payload item in lexographic order will be considered - all others are ignored. The
+// verifier returned by this method
+//
+// The presence of one or more config maps instructs the CVO to verify updates before they are
+// downloaded.
+//
+// The keys within the config map in the data field define how verification is performed:
+//
+// verifier-public-key-*: One or more GPG public keys in ASCII form that must have signed the
+//                        release image by digest.
+//
+// store-*: A URL (scheme file://, http://, or https://) location that contains signatures. These
+//          signatures are in the atomic container signature format. The URL will have the digest
+//          of the image appended to it as "<STORE>/<ALGO>=<DIGEST>/signature-<NUMBER>" as described
+//          in the container image signing format. The docker-image-manifest section of the
+//          signature must match the release image digest. Signatures are searched starting at
+//          NUMBER 1 and incrementing if the signature exists but is not valid. The signature is a
+//          GPG signed and encrypted JSON message. The file store is provided for testing only at
+//          the current time, although future versions of the CVO might allow host mounting of
+//          signatures.
+//
+// See https://github.com/containers/image/blob/ab49b0a48428c623a8f03b41b9083d48966b34a9/docs/signature-protocols.md
+// for a description of the signature store
+//
+// The returned verifier will require that any new release image will only be considered verified
+// if each provided public key has signed the release image digest. The signature may be in any
+// store and the lookup order is internally defined.
+func NewFromConfigMap(src string, data map[string]string, clientBuilder ClientBuilder) (*ReleaseVerifier, error) {
```

Comment:
this function is named ConfigMap but technically this doesn't care is the data came from one..?

Reply:
Maybe rename to `NewFromMap`?  The docstring also needs to be more generic, because this function doesn't care about looking up the configmap in a release's manifest set, it's just about creating a verifier from an already-found map.

Reply:
It uses the schema defined for a config map, even if it's coming from a map.  So the name is matching up with our config map rules, not any other set of rules.

Reply:
`NewFromConfigMapData` is probably the most accurate.